lib/widgets/gallery_view.dart

```diff
@@ -0,0 +1,258 @@
+                                ],
+                                child: Container(
+                                  decoration: BoxDecoration(
+                                      border: Border.all(
+                                        color: Colors.black,
+                                      ),
+                                      borderRadius: const BorderRadius.all(
+                                          Radius.circular(5))),
+                                  child: Container(
+                                    padding: const EdgeInsets.symmetric(
+                                        horizontal: 5.0, vertical: 3.0),
+                                    child: const Text('ADD',
+                                        textAlign: TextAlign.center,
+                                        style: TextStyle(
+                                            color: Color(0xFF000000))),
+                                  ),
+                                ),
+                              ))),
+                      Positioned(
+                          top: 25,
+                          right: 0,
+                          child: Padding(
+                              padding: const EdgeInsets.all(8.0),
+                              child: IconButton(
+                                onPressed: () {
+                                  addToLibrary();
+                                },
+                                icon: const Icon(Icons.favorite),
+                              )))
+                    ]),
+                    Padding(
+                        padding: const EdgeInsets.all(8),
+                        child: Align(
+                            alignment: Alignment.centerLeft,
+                            child: Text(
+                              '#' + tagList[index],
+                              style: TextStyle(
+                                  color: Colors.black.withOpacity(0.6)),
+                            ))),
+                  ],
+                ),
+              ));
+        });
+  }
+
+  // from https://stackoverflow.com/questions/60813379/how-to-put-searchbar-into-appbar-flutter
+  Widget buildSearchBar(BuildContext context, exerciseList) {
+    return Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: TextField(
+        decoration: const InputDecoration(hintText: 'Search Exercise'),
+        onChanged: (text) {
+          text = text.toLowerCase();
+          setState(() {
+            exercisesToDisplay = getDisplayItems(text, exerciseList);
+          });
+        },
+      ),
+    );
+  }
+
+  Widget build(BuildContext context) {
+    return Column(children: [
+      buildSearchBar(context, widget.exerciseList),
+      Expanded(
+          child: buildGrid(context, widget.exerciseList, widget.descriptionList,
+              widget.tagList, widget.playlistGallery))
+    ]);
+  }
+}
+
+class ExerciseDialog extends StatelessWidget {
```

Comment:
Could you move this to the same file as the `ExerciseCard` class mentioned above?